core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala

```diff
@@ -63,10 +63,15 @@ object JdkHttpClient {
     def convertRequest(req: Request[F]): Resource[F, HttpRequest] =
       flow.toPublisher(req.body.chunks.map(_.toByteBuffer)).evalMap { publisher =>
         convertHttpVersionFromHttp4s[F](req.httpVersion).map { version =>
+          def isStreaming = version match {
```

Comment:
To reduce confusion, I'd suggest renaming it. One might assume that `HTTP/2` mandates the streamed body or the presence of `Content-Length` to distinguish that the body is not streamed (while the streamed body alongside `Content-Length` is ok in HTTP/2). IMHO, something implying `consumeFully` might be fine.
```suggestion
          def consumeFully = version match {
```